src/ir_interpreter/interpret_ir.ml

```diff
@@ -586,14 +583,26 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
     k (interpret_fields env fs)
 
 and interpret_actor env ds fs k =
+    let self_id = (* HACK to locate self_id, better to add to ActorE *)
```

Comment:
Have you seen #1104? 😅

Reply:
I have my changes for the added name locally, can I commit to your branch?

Reply:
Done: 95c5ebfba

Reply:
https://github.com/dfinity/motoko/pull/1104 is a good find, but maybe a bit different since it binds the internal name of the actor rather than recording the external one. I wonder if we need to adapt freevars etc accordingly..